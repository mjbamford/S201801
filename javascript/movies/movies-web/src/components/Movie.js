import React from 'react'
import { Link } from 'react-router-dom'
import CommentList from './CommentList'

const Rating = ({ code, description }) => {
    return (<span>{code} {description}</span>)
}

const Movie = ({
    _id,
    title,
    yearReleased,
    rating,
    comments
}) => {
    return (<div>
        <Link to={`/movies/${_id}`}>{title}</Link>
        <span>{yearReleased}</span>
        <Rating {...rating} />
        <small>
            <Link to={`/movies/${_id}/edit`}>Edit</Link>
        </small>
        <CommentList items={comments} />
    </div>)
}

export default Movie