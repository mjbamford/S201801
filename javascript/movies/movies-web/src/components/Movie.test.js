import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Movie from './Movie'

describe('Movie', () => {
    const attrs = { _id: 666, title: 'Title', yearReleased: 1990 }
    const movie = renderer.create(
        <MemoryRouter>
            <Movie {...attrs} />
        </MemoryRouter>
    )
    it('displays the title and yearReleased', () => {
        expect(movie.toJSON()).toMatchSnapshot()
    })
})