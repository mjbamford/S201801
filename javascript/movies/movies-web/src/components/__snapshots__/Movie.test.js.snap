// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie displays the title and yearReleased 1`] = `
<div>
  <a
    href="/movies/666"
    onClick={[Function]}
  >
    Title
  </a>
  <span>
    1990
  </span>
  <span>
     
  </span>
  <small>
    <a
      href="/movies/666/edit"
      onClick={[Function]}
    >
      Edit
    </a>
  </small>
  <ul />
</div>
`;
